### ISSUE TITLE
Unknown Publisher

### ISSUE THREAD

**Issue by billmcilhargey** [nicehash/NiceHashMiner#5]
The NiceHashMinerLegacy exe is not code signed

**Comment by DillonN** [comment]
Yes, NHML will not be signed for the time being

**Comment by billmcilhargey** [comment]
ok - FYI - This will cause Windows Smart Screen to flag it and possibly Windows Defender

**Comment by DillonN** [comment]
Yes, since NHML is intended for experienced miners and includes 3rd party programs that will likely be flagged, it's expected that users will be able to make an exception in their AV.

**Comment by billmcilhargey** [comment]
Have to disable the real-time scanning of windows defender process as adding exceptions didnt help lol
All the other exes are code signed I noticed so when you do have time for future releases code signing the main one will greatly help ;)

But great software release its running so far so good!!!

**Comment by navjack** [comment]
right click on it and unblock

**Comment by billmcilhargey** [comment]
Correct that works  however it is still not trusted  so depending on the security restrictions of the system then it may or may not run or may run in a degraded performance while it is checking against the cloud

of course exceptions and turning down the securities would work too

Best practices now for production (again this is still in beta so thats another story :) ) software being released on Windows 10 is to release them code signed

**Comment by billmcilhargey** [comment]
Question (I know this thread is closed :) )

Are the miners located in bins  (not 3rd_party_bins)  compiled by nicehash then distributed?

If so then as part the this request down the road, add the code signing to the additional exes that you release and compile yourself  (I am working with the third parties to get them to release their code with digital signatures as well)

**Comment by billmcilhargey** [comment]
Think I answered my own question I found the script here that is used and would just need to be integrated into this new miner on compile

https://github.com/nicehash/NiceHashMinerLegacy/blob/master/sign_and_publish/sign_and_publish.bat

**Comment by billmcilhargey** [comment]
One more item to call out is when you run as an administrator on the app, UAC by default will ask you everytime are you sure you want to allow this app from an unknown publisher.  Where as with a trusted publisher it would allow it forward without having to say yes

**Comment by billmcilhargey** [comment]
any expectations on when code signing the main exe will happen or will not be able to happen with nicehash code signing cert?

@DillonN
